mlir/lib/Conversion/ArithToLLVM/ArithToLLVM.cpp

```diff
@@ -54,6 +54,25 @@ struct ConstrainedVectorConvertToLLVMPattern
   }
 };
 
+/// No-op bitcast. Propagate type input arg if converted source and dest types
+/// are the same.
+struct IdentityBitcastLowering final
```

Comment:
I think, to get the behavior you want here where identity bitcast on memref gets folded away before the general pattern for `arith.bitcast` kicks in, you want to set a `PatternBenefit`